### PR TITLE
Use a standalone test symlink for Lite

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of #14451

Also part of the standalone config was missing.

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @carmocca @akihironitta @borda @justusschock @awaelchli

### CODE REVIEW

File: .azure/gpu-tests-lite.yml [hunk 1]
```diff
@@ -78,7 +80,6 @@ jobs:
     - bash: bash run_standalone_tests.sh
       workingDirectory: tests/tests_lite
       env:
-        PL_USE_MOCKED_MNIST: "1"
```
Comment (carmocca) [inline .azure/gpu-tests-lite.yml:81]: I don't think Lite will use the MNIST dataset

File: tests/tests_lite/conftest.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import os
+from typing import List
+
+import pytest
+
+
+def pytest_collection_modifyitems(items: List[pytest.Function], config: pytest.Config) -> None:
```
Comment (carmocca) [inline tests/tests_lite/conftest.py:7]: This was missed, so the standalone test step is running all tests currently.

File: tests/tests_lite/conftest.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    """An adaptation of `tests/tests_pytorch/conftest.py::pytest_collection_modifyitems`"""
+    initial_size = len(items)
+    conditions = []
+    filtered, skipped = 0, 0
+
+    options = dict(
+        standalone="PL_RUN_STANDALONE_TESTS",
+        min_cuda_gpus="PL_RUN_CUDA_TESTS",
```
Comment (carmocca) [inline tests/tests_lite/conftest.py:15]: The only change is that I removed the slow test keyword

File: tests/tests_lite/conftest.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import os
+from typing import List
+
+import pytest
```
Comment (carmocca) [inline tests/tests_lite/conftest.py:4]: @awaelchli just as a heads up, you will probably need some of the other items we have in `conftest` for tests to work properly.

Comment (awaelchli) [reply tests/tests_lite/conftest.py:4]: Yes, I included these already in #14492

**Review by awaelchli — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Pull request merged**